### PR TITLE
Add scipioVersion to VersionFile

### PR DESCRIPTION
**Pull request by giginet**
Add scipioVersion key to VersionFile.
It helps to re-generate XCFrameworks when the compiler is updated.

### CODE REVIEW

File: Plugins/GenerateScipioVersion/plugin.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+@main
+struct GenerateScipioVersion: BuildToolPlugin {
+    func createBuildCommands(context: PluginContext, target: Target) async throws -> [Command] {
+        let zshPath = Path("/bin/zsh") // execute dummy command
+        let generatedSourceDir = context.pluginWorkDirectory
+        let generatedSourcePath = generatedSourceDir
+            .appending(subpath: "ScipioVersion.generated.swift")
+
+        let revision = try? fetchRepositoryVersion(context: context)
```
Comment (freddi-kit) [inline Plugins/GenerateScipioVersion/plugin.swift:12]: Why can we ignore this error by `try?`?

File: Plugins/GenerateScipioVersion/plugin.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            "rev-parse",
+            "HEAD",
+        ]
+        process.standardOutput = standardOutput
+
+        try process.run()
+        process.waitUntilExit()
+        let outputData = try standardOutput.fileHandleForReading.readToEnd()
+        guard let outputData else { return nil }
+        let revision = String(data: outputData, encoding: .utf8)?
```
Comment (freddi-kit) [inline Plugins/GenerateScipioVersion/plugin.swift:52]: How about `fatalError` if the revision is nil?

File: Plugins/GenerateScipioVersion/plugin.swift (hunk 1 above)
Comment (giginet) [reply Plugins/GenerateScipioVersion/plugin.swift:12]: This external command execution is in the build phase by a plugin.
So it fails silently. This is why to use `try?`

if failed, scipioVersion is not filled on VersionFile, `scipioVersion` is nullable.

File: Plugins/GenerateScipioVersion/plugin.swift (hunk 2 above)
Comment (giginet) [reply Plugins/GenerateScipioVersion/plugin.swift:52]: Same reason above

**Review by freddi-kit — APPROVED** [review]
LGTM, r? @giuk-jung

**Pull request merged**